docs/BackupNoSQL.md

```diff
@@ -0,0 +1,153 @@
+
+## Backup operations (create the **period** for restoration)
+
+1. Check the pod status before backup.
+
+   You must check the following four points using the `kubectl get pod` command before you start the backup operation.
+   * The number of Scalar product pods (You must compare this number with the number after the backup).
+   * The Scalar product pod names in the `NAME` column (You must compare these pod names with the pod names after the backup).
+   * The Scalar product pod status is `Running` in the `STATUS` column.
+   * The restart counts of each pod in the `RESTARTS` column (You must compare these restart counts with the restart counts after the backup).
+
+1. Pause Scalar Product pods using `scalar-admin`.
+
+   Please refer to the [How to use scalar-admin on a Kubernetes environment](./BackupNoSQL.md#how-to-use-scalar-admin-on-a-kubernetes-environment) section in this document for more details on the pause operation.
+
+1. Check the pause completed time.
+
+   You must note the pause completed time to decide the concrete **period** that you can restore data using the PITR feature.
+
+1. Take a backup using the backup feature of each database.
+
+   If you enable the automatic backup and PITR feature, all you have to do is wait for about 10 seconds since the managed databases take backup automatically. This 10 seconds period is the **period** that we can restore data by PITR.
+
+1. Unpause Scalar Product pods using `scalar-admin`.
+
+   Please refer to the [How to use scalar-admin on a Kubernetes environment](./BackupNoSQL.md#how-to-use-scalar-admin-on-a-kubernetes-environment) section in this document for more details on the unpause operation.
+
+1. Check the unpause started time.
+
+   You must note the unpause started time to decide the concrete **period** that you can restore data using the PITR feature.
+
+1. Check the pod status after backup.
+
+   You must check the following four points using the `kubectl get pod` command after you finish the backup operation.
+   * The number of Scalar product pods (You must compare this number with the number before the backup).
+   * The Scalar product pod names in the `NAME` column (You must compare these pod names with the pod names before the backup).
+   * The Scalar product pod status is `Running` in the `STATUS` column.
+   * The restart counts of each pod in the `RESTARTS` column (You must compare these restart counts with the restart counts before the backup).
+
+   **If the two values are different, you must re-try the backup operation all over again.** This is because, if some pods are added or restarted, those pods run with a `unpause` state. The `unpause` state pods cause data inconsistency in the backup data.
```

Comment:
@brfrn169  
Thank you for your review! (and sorry for the late response...)

I think the steps you pointed out are an ideal way, and I will create a new tool based on those steps.
However, in the current document, users deploy the `scalar-admin` pod, and it returns the **unpause started time** in the stdout.
It is a bit confusing and I think there is a possibility that users will make misunderstand the unpause started time as follows.

1. Check the pod status before backup.
1. Pause Scalar Product pods using `scalar-admin`.
1. Check the pause completed time. <- Users can do this step based on the output of `scalar-admin` pod.
1. Wait for about 10 seconds.
1. Check the unpause started time. <- Users cannot use the output of `scalar-admin` pod.  They need to check this time using their own clock.
1. Check the pod status after backup.
1. === Here is a time lag ===
1. Unpause Scalar Product pods using `scalar-admin`. <- The `scalar-admin` pod outputs the **unpause started time**, but users cannot use this time to decide the **period** since this time is later than status checking.

In these steps, users need to decide the **period** based on the `step 3` and the `step 5`.
In other words, there is a possibility that the pods are created or restarted at the point of `step 7` in the above steps. This causes backup data inconsistency.
However, when users run the unpause requests at `step 8`, the `scalar-admin` pod outputs **unpause started time** in the stdout.
If users pick up this time as a unpause started time (this is a wrong decision), there is a possibility that the backup data inconsistency occurs.
This output is confusing, but there is no way to disable outputting **unpause started time** in the stdout.

As the above, to avoid making misunderstandings of users, I think it is better to run unpause requests before checking pod status and check the unpause started time based on the output of `scalar-admin` pod.
I think this step also can keep the consistency of backup data since the **pause completed time** and **unpause started time** (i.e., the **period**) exist between the two status checking operations.

What do you think?

Reply:
Sounds good. Thank you for the explanation!